### PR TITLE
feature(workspace): Add support for custom monitoring settings for de…

### PR DESCRIPTION
**Pull request by ioanpocol**
…sks and custom workspaces

### CODE REVIEW

File: client/app/scripts/superdesk-desks/aggregate.js [hunk 1]
```diff
@@ -54,19 +54,80 @@
...
+                    this.loading = false;
                 }));
         }));
 
         /**
+         * Read the setting for current selected workspace(desk or custom workspace)
+         * If the current selected workspace is a desk the settings are read from desk
+         * If the current selected workspace is a custom workspace the settings are read from
+         * user preferences
+         * @return {list} - the list of settings
```
Comment (plamut) [inline client/app/scripts/superdesk-desks/aggregate.js]: Doesn't this method return a _promise_ which is resolved with a list? The docstring is a bit misleading here...

File: client/app/scripts/superdesk-search/search.js [hunk 2]
```diff
@@ -1329,11 +1330,13 @@
...
         this.spikeItems = function() {
             spike.spikeMultiple(multi.getItems());
+            $rootScope.$broadcast('item:spike');
             multi.reset();
         };
 
         this.unspikeItems = function() {
             spike.unspikeMultiple(multi.getItems());
+            $rootScope.$broadcast('item:unspike');
             multi.reset();
```
Comment (plamut) [inline client/app/scripts/superdesk-search/search.js]: Docstrings should be added to the both changed methods.

**Comment by plamut** [comment]
Please check if the tests indeed sufficiently cover the changes and edge cases (could not verify that), though the things look good.

:+1:

**Comment by eozan** [comment]
:+1:

File: client/app/scripts/superdesk-desks/aggregate.js [hunk 3]
```diff
@@ -54,19 +54,80 @@
...
+                });
+        }
+
+        /**
+         * Read the settings when the current workspace
+         * selection is changed
+         */
+        $scope.$watch(function() {
+            return workspaces.active;
+        }, refresh, true);
```
Comment (petrjasek) [inline client/app/scripts/superdesk-desks/aggregate.js]: is there a need for recursive watch?

File: client/app/scripts/superdesk-desks/aggregate.js [hunk 4]
```diff
@@ -54,19 +54,80 @@
...
+
+        /**
+         * Read the setting for current workspace after
+         * the settings has been change
+         */
+        this.refreshGroups = function() {
```
Comment (petrjasek) [inline client/app/scripts/superdesk-desks/aggregate.js]: you can just assign refresh here
